### DISCUSSION TITLE
gltf is rendered without texture

### DISCUSSION THREAD

**Discussion by xbz-24** [bevyengine/bevy#12901] [Q&A]
![image](https://github.com/bevyengine/bevy/assets/68678258/afb1f3b6-8c1f-4d92-afd9-64bccbc6066c)
this is my project structure
![image](https://github.com/bevyengine/bevy/assets/68678258/a9b741d3-ca61-42dd-a77c-944b63a47bc5)
```rust
    commands.spawn(SceneBundle{
        scene: asset_server.load("gltf/scene.gltf#Scene0"),
        ..default()
    });

```

**Comment by xbz-24** [comment]
gltf/scene.gltf is the scene its loading, and its correct but I need this whole whole model to use this aqua_diffuse.png


like this


![image](https://github.com/bevyengine/bevy/assets/68678258/eaa5ea0b-6a2f-4c2b-a18e-f42dca904123)

**Comment by xbz-24** [comment]
im just doing in main:



```rust

fn main() {
    let mut app = App::new();
    app.add_plugins(DefaultPlugins);
    #[cfg(debug_assertions)]
    {
        use bevy::diagnostic::FrameTimeDiagnosticsPlugin;
        app.add_plugins(FrameTimeDiagnosticsPlugin::default());
    }
    app.add_systems(Startup, setup);
    app.add_systems(Update, camera_movement);
    app.run();
}

**Comment by xbz-24** [comment]
and in setup():


```rust
fn setup(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut materials: ResMut<Assets<StandardMaterial>>,
    mut meshes: ResMut<Assets<Mesh>>,
) {
    commands.spawn(Camera3dBundle {
        transform: Transform::from_xyz(3.0, 3.0, 3.0).looking_at(Vec3::ZERO, Vec3::Y),
        ..std::default::Default::default()
    });

    commands.spawn(SceneBundle{
        scene: asset_server.load("gltf/scene.gltf#Scene0"),
        ..default()
    });

```
among other lines of code but this is the only ones that refer to the gltf

**Comment by xbz-24** [comment]
please help


also, when I try either exporting this, as glb or gltf (with the textures mapped) and I try to load that exported file with the asset_server

this is what I get


![image](https://github.com/bevyengine/bevy/assets/68678258/c64b9ed4-2895-4c57-a110-5b7e83f88d3d)


like only the fan gets rendered

at least with the correct textures tho

but only the fan out of the whole scene/mesh

**Comment by xbz-24** [comment]
also if I try to call for Scene1 instead of Scene0 I get this

![image](https://github.com/bevyengine/bevy/assets/68678258/46f8e2f1-6edc-403b-82c7-42f090e142b7)


this is the message from the console output: https://pastebin.com/m1xHr93u

**Comment by xbz-24** [comment]
please let me know how to put textures to this model

![image](https://github.com/bevyengine/bevy/assets/68678258/df5fc2ce-5973-4827-9cd3-060d8885238f)

I am so close to actually mapping the textures but I am not sure how


please help

**Comment by xbz-24** [comment]
https://github.com/Xbz-24/Bevy-Sandbox.git


this is the code if needed.

**Comment by xbz-24** [comment]
please

can

someone

take a look


at why is


this


not mapping the textures properly?

**Comment by torsteingrindvik** [comment]
I cloned your repo and used this: https://gltf-viewer.donmccurdy.com/

I dropped the entire `gltf` folder into the validator such that it has access to all files.

The viewer shows the same as Bevy: The meshes are there but the textures are not attached.

I think it's more likely you are having Blender exporting issues than Bevy issues.

My advice then is to figure out the Blender side of things first, validate (with the one I linked or any GLTF validator) that things are working, _then_ try it with Bevy.

**Comment by xbz-24** [comment]
but how do I do that without requiring of blender?
I want to do that through code, I was thinking maybe uisng a PbrBundle take as mesh the gltf model and using a standardmaterial of the aqua_diffuse.png  as the material the entire #Scene0 from the .gltf needs to be mapped with the same png

**Comment by xbz-24** [comment]
I dont want to do it in blender, I know this model is not PBR so if I try to use a PBR bundle with the material as the texture of aqua_diffuse.png do you think i twill work?

**Comment by xbz-24** [comment]
![image](https://github.com/bevyengine/bevy/assets/68678258/4798cca7-3d7c-44cb-b58e-412d42e091c7)
thi is the content of my aqua_emissive.png
this is for aqua_diffuse.png
![image](https://github.com/bevyengine/bevy/assets/68678258/83d72bb3-7782-409d-a6be-61f43f05938f)

**Comment by xbz-24** [comment]
I am loading the emissive png using asset server
then I add it as a Standard material to my ResMut<Assets<StandardMaterial>>
how do I use it though
```rust
pub fn setup(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut materials: ResMut<Assets<StandardMaterial>>,
    mut meshes: ResMut<Assets<Mesh>>,
    mut window_query: Query<&mut Window, With<PrimaryWindow>>,
) {
    commands.spawn(Camera3dBundle {
        transform: Transform::from_xyz(3.0, 3.0, 3.0).looking_at(Vec3::ZERO, Vec3::Y),
        ..std::default::Default::default()
    }).insert(Player::default());

    let emissive_texture_handle = asset_server.load("assets/gltf/textures/aqua_emissive.png");

    let emissive_material = materials.add(StandardMaterial {
        emissive_texture: Some(emissive_texture_handle.clone()),
        emissive: Color::WHITE, 
        ..default()
    });


    commands.spawn(SceneBundle{
        scene: asset_server.load("gltf/scene.gltf#Scene0"),
        ..default()
    });
```


for example SceneBundle does not receive a material
should I use a PbrBundle then?


if I use PbrBundle then it doesnt render at all anything
and it doesnt even say error log
look 

![image](https://github.com/bevyengine/bevy/assets/68678258/9a0954ac-edb5-4ff0-b7f2-cfc6c17b8435)

```rust
pub fn setup(
    mut commands: Commands,
    asset_server: Res<AssetServer>,
    mut materials: ResMut<Assets<StandardMaterial>>,
    mut meshes: ResMut<Assets<Mesh>>,
    mut window_query: Query<&mut Window, With<PrimaryWindow>>,
) {
    commands.spawn(Camera3dBundle {
        transform: Transform::from_xyz(3.0, 3.0, 3.0).looking_at(Vec3::ZERO, Vec3::Y),
        ..std::default::Default::default()
    }).insert(Player::default());

    let emissive_texture_handle = asset_server.load("gltf/textures/aqua_emissive.png");

    let emissive_material = materials.add(StandardMaterial {
        emissive_texture: Some(emissive_texture_handle.clone()),
        emissive: Color::WHITE, 
        ..default()
    });


    commands.spawn(PbrBundle{
        mesh: asset_server.load("gltf/scene.gltf#Scene0"),
        material: emissive_material,
        ..default()
    });

```

**Comment by xbz-24** [comment]
this is so frustrating please help me fix without blender please

**Comment by xbz-24** [comment]
@torsteingrindvik

**Comment by suprohub** [comment]
any updates?
